File: platform/robot/src/navigation/maneuvers/parking/geometry.py
"""Angle normalisation helper for the parking controller."""

from __future__ import annotations

import math


def normalise_angle(angle: float) -> float:
    """Wrap an angle into the ``(-pi, pi]`` range."""
    while angle > math.pi:
        angle -= 2 * math.pi
    while angle < -math.pi:
        angle += 2 * math.pi
    return angle

File: platform/robot-go/internal/nav/parking/geometry.go
package parking

import "math"

// NormaliseAngle wraps an angle into the (-pi, pi] range.
func NormaliseAngle(angle float64) float64 {
	for angle > math.Pi {
		angle -= 2 * math.Pi
	}
	for angle < -math.Pi {
		angle += 2 * math.Pi
	}
	return angle
}
